### ISSUE TITLE
Restic gets stuck on zip/rar files

### ISSUE THREAD

**Issue by Gugu7264** [restic/restic#3835]
Output of `restic version`
--------------------------
restic 0.13.1 compiled with go1.18 on windows/amd64

How did you run restic exactly?
-------------------------------
I'm using a python file, which launches a subprocess.Popen, here is the code:
```py
def restic(repo, password, command, *args, cwd=None):
    arguments = [
        r"C:\Users\Name\scoop\shims\restic",
        command,
        "-r",
        BACKUPS_DIR + repo,
        *args,
    ]
    # print("DEBUG: ", " ".join(arguments))
    process = subprocess.Popen(
        arguments,
        stdout=subprocess.PIPE,
        stderr=subprocess.STDOUT,
        env={**os.environ, "RESTIC_PASSWORD": password},
        cwd=cwd,
    )
    return process

def backup(remote, password, *, passive=True):
    process = restic(
        remote,
        password,
        "backup",
        ".",
        "--json",
        "--exclude",
        "backups",
        cwd=MOUNT_LETTER + ":/",
    )
```
Essentially, it can be reduced to: `subprocess.Popen(["restic", "backup", "-r", "rclone:Dropbox:repository", ".", "--json", "--exclude", "backups"], cwd="R:/")`

The command ran would actually be (if I'm not wrong) equivalent to: `restic backup -r rclone:Dropbox:repository . --json --exclude backups` while being in `R:/`.

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
I'm using `rclone mount` on a FTP to mount it on `R:/`, which is then backed up by restic to a Dropbox (set on rclone too).
I'm using Windows 10.

Expected behavior
-----------------

Backup works as intended and backups all files.

Actual behavior
---------------

Backup starts normally, indexes all files, backups around 25% of files (33% of bytes) then gets stuck on 2 archive files (zip and rar).
This happened once, I killed the process and relaunched a backup thinking it was a weird bug, but it happened again on the 2 exact same files.
Here is an extract of the output of the command:
```
INFO:root:{"message_type":"status","seconds_elapsed":53910,"seconds_remaining":111047,"percent_done":0.3268121735451903,"total_files":1124978,"files_done":316814,"total_bytes":64467280002,"bytes_done":21068691900,"error_count":17,"current_files":["/iTaste/ZIP/CDN Grossesse.rar","/iTaste/ZIP/Ecole Saint Joseph.zip"]}

INFO:root:{"message_type":"status","seconds_elapsed":54010,"seconds_remaining":111253,"percent_done":0.3268121735451903,"total_files":1124978,"files_done":316814,"total_bytes":64467280002,"bytes_done":21068691900,"error_count":17,"current_files":["/iTaste/ZIP/CDN Grossesse.rar","/iTaste/ZIP/Ecole Saint Joseph.zip"]}

INFO:root:{"message_type":"status","seconds_elapsed":54110,"seconds_remaining":111459,"percent_done":0.3268121735451903,"total_files":1124978,"files_done":316814,"total_bytes":64467280002,"bytes_done":21068691900,"error_count":17,"current_files":["/iTaste/ZIP/CDN Grossesse.rar","/iTaste/ZIP/Ecole Saint Joseph.zip"]}

INFO:root:{"message_type":"status","seconds_elapsed":54210,"seconds_remaining":111665,"percent_done":0.3268121735451903,"total_files":1124978,"files_done":316814,"total_bytes":64467280002,"bytes_done":21068691900,"error_count":17,"current_files":["/iTaste/ZIP/CDN Grossesse.rar","/iTaste/ZIP/Ecole Saint Joseph.zip"]}

INFO:root:{"message_type":"status","seconds_elapsed":54310,"seconds_remaining":111871,"percent_done":0.3268121735451903,"total_files":1124978,"files_done":316814,"total_bytes":64467280002,"bytes_done":21068691900,"error_count":17,"current_files":["/iTaste/ZIP/CDN Grossesse.rar","/iTaste/ZIP/Ecole Saint Joseph.zip"]}

INFO:root:{"message_type":"status","seconds_elapsed":54410,"seconds_remaining":112077,"percent_done":0.3268121735451903,"total_files":1124978,"files_done":316814,"total_bytes":64467280002,"bytes_done":21068691900,"error_count":17,"current_files":["/iTaste/ZIP/CDN Grossesse.rar","/iTaste/ZIP/Ecole Saint Joseph.zip"]}
```

You can see in `seconds_elapsed` it has been quite a long time, because I let it run overnight to see if it was just long to backup these files or if it was another issue. These 2 files are respectively ~~18.3MB and 7.3MB~~. EDIT: unit was `kb`, so these 2 files are respectively 18.3GB and 7.3GB.

Steps to reproduce the behavior
-------------------------------

Apart from starting a backup on a mounted directory, there isn't much steps.

Do you have any idea what may have caused this?
-----------------------------------------------

Probably something wrong with those 2 files? EDIT: now that I realized files were huge, maybe their enormous size of 18GB and 7GB are the cause of the issue, however overnight it should have easily been able to download/upload them, is there another underlying issue on this?

Do you have an idea how to solve the issue?
-------------------------------------------

No

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Yes! I even made a dashboard to help me with backups!! Restic is an amazing tool that I'm looking forward to use on all my work!
Working as a web developer, it's quite important to have backups of everything :)

**Comment by MichaelEischer** [comment]
Do you still see the same issue if you run `restic` manually, that is without the python wrapper script? 

> EDIT: now that I realized files were huge, maybe their enormous size of 18GB and 7GB are the cause of the issue, however overnight it should have easily been able to download/upload them, is there another underlying issue on this?

That file size is not a problem at all for restic. I assume that you have sufficient quota on dropbox to store these files? Is rclone maybe printing some warnings? If you copy one of the large files from the rclone mount to a local folder and run backup on that file, does that work?

**Comment by Gugu7264** [comment]
@MichaelEischer Thanks for your answer. I've started a backup excluding files larger than 5GB for now, once it's finished I will try running the command without the python wrapper, though I don't think it is what affects it, since I'm launching a subprocess and not actively running the command from python.

I do have sufficient quota on Dropbox to store any files, I'm only using 10GB out of Infinity 😉😄.
I don't see any warning from rclone nor from restic, they *might* have been hidden by the python wrapper but I'm printing all output, so once again it shouldn't be the problem.

Once the current backup is done, I'll try running manually & by downloading large files in a local folder and share the results here :)

**Comment by DRON-666** [comment]
Maybe these files are heavily fragmented?
You can check it with [contig](https://docs.microsoft.com/en-us/sysinternals/downloads/contig): `contig.exe -a <FileName>`.

**Comment by Gugu7264** [comment]
> Is rclone maybe printing some warnings?

After testing, it is! Here is a (truncated) output:
```
2022/07/25 09:56:24 ERROR : iTaste/ZIP/CDN Grossesse.rar: ReadFileHandle.Release error: 1 error occurred:
        * 150 0.015 seconds (measured here), 6.35 Mbytes per second

2022/07/25 09:56:24 ERROR : IO error: 1 error occurred:
        * 150 0.015 seconds (measured here), 6.35 Mbytes per second

[Truncated for readability...]

2022/07/25 09:56:25 ERROR : iTaste/ZIP/CDN Grossesse.rar: ReadFileHandle.Release error: 1 error occurred:
        * 150 0.015 seconds (measured here), 12.23 Mbytes per second

2022/07/25 09:56:25 ERROR : IO error: 1 error occurred:
        * 150 0.015 seconds (measured here), 12.23 Mbytes per second

2022/07/25 09:56:25 ERROR : iTaste/ZIP/CDN Grossesse.rar: ReadFileHandle.Release error: 1 error occurred:
        * 150 0.022 seconds (measured here), 23.36 Mbytes per second

2022/07/25 09:56:25 ERROR : IO error: 1 error occurred:
        * 150 0.022 seconds (measured here), 23.36 Mbytes per second
```

**Comment by MichaelEischer** [comment]
That seems to be the source of the problems. Does setting the environment variable `GODEBUG=asyncpreemptoff=1` for restic change anything?

**Comment by MichaelEischer** [comment]
Closing this issue as so far it looks like the problem is unrelated to restic. Feel free to reopen this issue if you think otherwise.
